appliances/MinIO/appliance.sh

```diff
@@ -0,0 +1,331 @@
+# This may improve systemctl setups where other services use `After=minio.server`
+# Uncomment the line to enable the functionality
+# Type=notify
+
+# Let systemd restart this service always
+Restart=always
+
+# Specifies the maximum file descriptor number that can be opened by this process
+LimitNOFILE=65536
+
+# Specifies the maximum number of threads this process can create
+TasksMax=infinity
+
+# Disable timeout logic and wait until process is stopped
+TimeoutStopSec=infinity
+SendSIGKILL=no
+
+[Install]
+WantedBy=multi-user.target
+
+# Built for ${project.name}-${project.version} (${project.name})
+
+EOF
+
+    msg info "Enable systemd service for MinIO"
+    systemctl enable minio.service
+}
+
+create_default_environment_variable_file()
+{
+    msg info "Create default Environment Variable File for MinIO"
+    cat > /etc/default/minio <<'EOF'
+#
+# ATTENTION: This file is managed by MinIO Service Contextualization. DO NOT manually edit this file, instead
+# use contextualization parameters to update MinIO attributes.
+#
+# Begin Environment Variable File
+#
+# MINIO_ROOT_USER and MINIO_ROOT_PASSWORD sets the root account for the MinIO server.
+# This user has unrestricted permissions to perform S3 and administrative API operations on any resource in the deployment.
+# Omit to use the default values 'minioadmin:minioadmin'.
+# MinIO recommends setting non-default values as a best practice, regardless of environment
+
+MINIO_ROOT_USER=myminioadmin
+MINIO_ROOT_PASSWORD=minio-secret-key-change-me
+
+# MINIO_VOLUMES sets the storage volume or path to use for the MinIO server.
+
+MINIO_VOLUMES="/mnt/data"
+
+# MINIO_OPTS sets any additional commandline options to pass to the MinIO server.
+# For example, `--console-address :9001` sets the MinIO Console listen port
+MINIO_OPTS="--console-address :9001"
+
+# MINIO_SERVER_URL sets the hostname of the local machine for use with the MinIO Server
+# MinIO assumes your network control plane can correctly resolve this hostname to the local machine
+
+# Uncomment the following line and replace the value with the correct hostname for the local machine and port for the MinIO server (9000 by default).
+
+#MINIO_SERVER_URL="http://minio.example.net:9000"
+EOF
+}
+
+update_environment_variable_file()
+{
+    msg info "Update MinIO root user"
+    sed -i "s/^MINIO_ROOT_USER.*/MINIO_ROOT_USER=${ONEAPP_MINIO_ROOT_USER}/g" /etc/default/minio
+    msg info "Update MinIO root user password"
+    sed -i "s/^MINIO_ROOT_PASSWORD.*/MINIO_ROOT_PASSWORD=${ONEAPP_MINIO_ROOT_PASSWORD}/g" /etc/default/minio
+    msg info "Update MinIO volumes"
+    sed -i "s|^MINIO_VOLUMES.*|MINIO_VOLUMES=\"${1}\"|g" /etc/default/minio
+    msg info "Update MinIO opts"
+    sed -i "s/^MINIO_OPTS.*/MINIO_OPTS=\"${ONEAPP_MINIO_OPTS}\"/g" /etc/default/minio
+
+    msg info "Restart minio service"
+    systemctl restart minio
+}
+
+postinstall_cleanup()
+{
+    msg info "Delete cache and stored packages"
+    apt-get autoclean
```

Comment:
When working with apt-get in scripts it's good to set `export DEBIAN_FRONTEND=noninteractive` somewhere on the top of the script.